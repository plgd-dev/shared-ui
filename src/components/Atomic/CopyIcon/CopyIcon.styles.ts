import { css } from '@emotion/react'

import { get, ThemeType } from '../_theme'

export const copy = (theme: ThemeType) => css`
    display: inline-flex;
    color: ${get(theme, `CopyIcon.color`)};
    transition: all 0.3s;
    margin-left: 8px;
    cursor: pointer;

    &:hover {
        color: ${get(theme, `CopyIcon.hover.color`)};
    }
`
