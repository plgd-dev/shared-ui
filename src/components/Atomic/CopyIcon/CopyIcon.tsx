import React, { FC } from 'react'

import * as styles from '../SimpleStripTable/SimpleStripTable.styles'
import IconCopy from '../Icon/components/IconCopy'
import { copyToClipboard } from '../../../common/utils'
import Tooltip from '../Tooltip'

const CopyIcon: FC<any> = (props) => {
    const { i18n, value } = props

    return (
        <Tooltip content={i18n.content} css={styles.copy}>
            <IconCopy onClick={() => copyToClipboard(value)} />
        </Tooltip>
    )
}

CopyIcon.displayName = 'CopyIcon'

export default CopyIcon
