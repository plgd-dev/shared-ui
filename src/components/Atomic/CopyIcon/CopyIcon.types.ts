export type Props = {
    i18n: {
        content: string
    }
    value: string | number
    dataTestId?: string
}
