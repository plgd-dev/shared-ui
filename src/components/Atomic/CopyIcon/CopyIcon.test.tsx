import { fireEvent } from '@testing-library/react'
import { forEach } from 'lodash'

import { render } from '../../../../test/jest-wrapper'
import CopyIcon from './CopyIcon'

describe('CopyIcon', () => {
    it('renders without crashing', () => {
        const { asFragment } = render(<CopyIcon i18n={{ content: 'Copy' }} value='Test value' />)
        expect(asFragment()).toMatchSnapshot()
    })

    it('calls copyToClipboard function on click', () => {
        const copyToClipboardMock = jest.spyOn(require('../../../common/utils/copy-to-clipboard'), 'copyToClipboard')

        const { container } = render(<CopyIcon dataTestId='test-icon' i18n={{ content: 'Copy' }} value='Test value' />)
        const buttons = container.querySelectorAll('[data-test-id="test-icon"]')

        forEach(buttons, (button) => {
            fireEvent.click(button)
        })

        expect(copyToClipboardMock).toHaveBeenCalledWith('Test value')
    })
})
