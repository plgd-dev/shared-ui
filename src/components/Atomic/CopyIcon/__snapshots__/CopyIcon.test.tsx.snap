// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CopyIcon renders without crashing 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-transition: all 0.3s;
  transition: all 0.3s;
  margin-left: 8px;
  cursor: pointer;
}

<div
    class="emotion-0"
  >
    <svg
      fill="none"
      height="16"
      viewBox="0 0 24 24"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M10.75 2a2.5 2.5 0 0 0-2.5 2.5v8.75c0 1.379 1.121 2.5 2.5 2.5h8.75c1.379 0 2.5-1.121 2.5-2.5V4.5C22 3.121 20.879 2 19.5 2zM4.5 8.25a2.5 2.5 0 0 0-2.5 2.5v8.75C2 20.879 3.121 22 4.5 22h8.75c1.379 0 2.5-1.121 2.5-2.5V17h-2.5v2.5H4.5v-8.75H7v-2.5z"
        fill="currentcolor"
      />
    </svg>
  </div>
</DocumentFragment>
`;
